modules/stochastic_tools/include/utils/GaussianProcessHandler.h

```diff
@@ -52,11 +52,15 @@ class GaussianProcessHandler
    * @param tao_options Additional options if TAO is used for parameter optimization.
    * @param show_tao Switch to show details of TAO optimization.
    */
-  void setupCovarianceMatrix(const RealEigenMatrix & training_params,
-                             const RealEigenMatrix & training_data,
-                             MooseEnum opt_type,
-                             std::string tao_options = "",
-                             bool show_tao = false);
+  void
+  setupCovarianceMatrix(RealEigenMatrix training_params, // const RealEigenMatrix & training_params
+                        RealEigenMatrix training_data,
+                        MooseEnum opt_type,
```

Comment:
At this point I think we could add a struct that contains (maybe with a name of GPOptimizationOptions or similar):
opt_type
tao_options
iter_adam
batch size
leraning_rate

Reply:
Looks like this needs changes to both Adam and TAO usage. Can we address this in a separate issue?